Autogenerated tests failing in `python-bigquery-storage`

See PR https://github.com/googleapis/python-bigquery-storage/pull/742 where the autogenerated tests are failing in the `Kokoro Prerelease Dependencies ` presubmit. See the stack trace below:

```
            # Test the case when there is a universe mismatch from the client.
            #
            # TODO: Make this test unconditional once the minimum supported version of
            # google-api-core becomes 2.15.0 or higher.
>           api_core_major, api_core_minor, _ = [
                int(part) for part in api_core_version.__version__.split(".")
            ]

tests/unit/gapic/bigquery_storage_v1beta2/test_big_query_read.py:342: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = 

    api_core_major, api_core_minor, _ = [
>       int(part) for part in api_core_version.__version__.split(".")
    ]
E   ValueError: invalid literal for int() with base 10: '0rc0'

tests/unit/gapic/bigquery_storage_v1beta2/test_big_query_read.p
```
The `Kokoro Prerelease Dependencies ` tests autogenerated code against pre-release versions of dependencies. The issue started when `google-api-core` `2.17.0rc0` was released. We need to update this code to work with rc versions.

https://pypi.org/project/google-api-core/2.17.0rc0/

See related issue https://github.com/googleapis/gapic-generator-python/issues/1791

--- a/.github/sync-repo-settings.yaml
+++ b/.github/sync-repo-settings.yaml
@@ -49,6 +49,7 @@ branchProtectionRules:
     - 'showcase-unit-add-iam-methods'
     - 'integration'
     - 'goldens-lint'
+    - 'goldens-prerelease'
     - 'style-check'
     - 'snippetgen'
     - 'unit (3.7)'

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -326,6 +326,27 @@ jobs:
           nox -f tests/integration/goldens/eventarc/noxfile.py -s mypy-3.12 blacken lint
           nox -f tests/integration/goldens/logging/noxfile.py -s mypy-3.12 blacken lint
           nox -f tests/integration/goldens/redis/noxfile.py -s mypy-3.12 blacken lint
+  goldens-prerelease:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: 'pip'
+      - name: Install nox.
+        run: |
+            python -m pip install nox
+      - name: Run the prerelease_deps nox session
+        # Exclude testing for asset which requires dependency google-cloud-org-policy
+        # in order to run unit tests
+        # See https://github.com/googleapis/gapic-generator-python/issues/1806
+        run: |
+          nox -f tests/integration/goldens/credentials/noxfile.py -s prerelease_deps
+          nox -f tests/integration/goldens/eventarc/noxfile.py -s prerelease_deps
+          nox -f tests/integration/goldens/logging/noxfile.py -s prerelease_deps
+          nox -f tests/integration/goldens/redis/noxfile.py -s prerelease_deps
   style-check:
     runs-on: ubuntu-latest
     steps:

--- a/gapic/templates/noxfile.py.j2
+++ b/gapic/templates/noxfile.py.j2
@@ -4,6 +4,7 @@
 
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
@@ -38,6 +39,7 @@ nox.sessions = [
     "docs",
     "blacken",
     "lint",
+    "prerelease_deps",
 ]
 
 @nox.session(python=ALL_PYTHON)
@@ -46,6 +48,80 @@ def unit(session):
 
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
     session.install('-e', '.')
+
+    session.run(
+        'py.test',
+        '--quiet',
+        '--cov={{ api.naming.module_namespace|join("/") }}/{{ api.naming.versioned_module_name }}/',
+        '--cov=tests/',
+        '--cov-config=.coveragerc',
+        '--cov-report=term',
+        '--cov-report=html',
+        os.path.join('tests', 'unit', ''.join(session.posargs))
+    )
+
+@nox.session(python=ALL_PYTHON[-1])
+def prerelease_deps(session):
+    """Run the unit test suite."""
+
+    # Install test environment dependencies
+    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
+
+    # Install the package without dependencies
+    session.install('-e', '.', '--no-deps')
+
+    # Because we test the minimum dependency versions using the minimum Python
+    # version, the lowest python runtime that we test has a corresponding constraints
+    # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
+    # dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{ALL_PYTHON[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    constraints_deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    session.install(*constraints_deps)
+
+    prerel_deps = [
+        "googleapis-common-protos",
+        "google-api-core",
+        "google-auth",
+        "grpcio",
+        "grpcio-status",
+        "protobuf",
+        "proto-plus",
+    ]
+
+    for dep in prerel_deps:
+        session.install("--pre", "--no-deps", "--upgrade", dep)
+
+    # Remaining dependencies
+    other_deps = [
+        "requests",
+    ]
+    session.install(*other_deps)
+
+    # Print out prerelease package versions
+
+    session.run("python", "-c", "import google.api_core; print(google.api_core.__version__)")
+    session.run("python", "-c", "import google.auth; print(google.auth.__version__)")
+    session.run("python", "-c", "import grpc; print(grpc.__version__)")
+    session.run(
+        "python", "-c", "import google.protobuf; print(google.protobuf.__version__)"
+    )
+    session.run(
+        "python", "-c", "import proto; print(proto.__version__)"
+    )
 
     session.run(
         'py.test',

--- a/gapic/templates/noxfile.py.j2
+++ b/gapic/templates/noxfile.py.j2
@@ -62,7 +62,7 @@ def unit(session):
 
 @nox.session(python=ALL_PYTHON[-1])
 def prerelease_deps(session):
-    """Run the unit test suite."""
+    """Run the unit test suite against pre-release versions of dependencies."""
 
     # Install test environment dependencies
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
@@ -70,8 +70,8 @@ def prerelease_deps(session):
     # Install the package without dependencies
     session.install('-e', '.', '--no-deps')
 
-    # Because we test the minimum dependency versions using the minimum Python
-    # version, the lowest python runtime that we test has a corresponding constraints
+    # We test the minimum dependency versions using the minimum Python
+    # version so the lowest python runtime that we test has a corresponding constraints
     # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
     # dependencies and extras.
     with open(

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -225,7 +225,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -241,7 +241,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/asset/noxfile.py
+++ b/tests/integration/goldens/asset/noxfile.py
@@ -15,6 +15,7 @@
 #
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
@@ -49,6 +50,7 @@ nox.sessions = [
     "docs",
     "blacken",
     "lint",
+    "prerelease_deps",
 ]
 
 @nox.session(python=ALL_PYTHON)
@@ -57,6 +59,80 @@ def unit(session):
 
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
     session.install('-e', '.')
+
+    session.run(
+        'py.test',
+        '--quiet',
+        '--cov=google/cloud/asset_v1/',
+        '--cov=tests/',
+        '--cov-config=.coveragerc',
+        '--cov-report=term',
+        '--cov-report=html',
+        os.path.join('tests', 'unit', ''.join(session.posargs))
+    )
+
+@nox.session(python=ALL_PYTHON[-1])
+def prerelease_deps(session):
+    """Run the unit test suite."""
+
+    # Install test environment dependencies
+    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
+
+    # Install the package without dependencies
+    session.install('-e', '.', '--no-deps')
+
+    # Because we test the minimum dependency versions using the minimum Python
+    # version, the lowest python runtime that we test has a corresponding constraints
+    # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
+    # dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{ALL_PYTHON[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    constraints_deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    session.install(*constraints_deps)
+
+    prerel_deps = [
+        "googleapis-common-protos",
+        "google-api-core",
+        "google-auth",
+        "grpcio",
+        "grpcio-status",
+        "protobuf",
+        "proto-plus",
+    ]
+
+    for dep in prerel_deps:
+        session.install("--pre", "--no-deps", "--upgrade", dep)
+
+    # Remaining dependencies
+    other_deps = [
+        "requests",
+    ]
+    session.install(*other_deps)
+
+    # Print out prerelease package versions
+
+    session.run("python", "-c", "import google.api_core; print(google.api_core.__version__)")
+    session.run("python", "-c", "import google.auth; print(google.auth.__version__)")
+    session.run("python", "-c", "import grpc; print(grpc.__version__)")
+    session.run(
+        "python", "-c", "import google.protobuf; print(google.protobuf.__version__)"
+    )
+    session.run(
+        "python", "-c", "import proto; print(proto.__version__)"
+    )
 
     session.run(
         'py.test',

--- a/tests/integration/goldens/asset/noxfile.py
+++ b/tests/integration/goldens/asset/noxfile.py
@@ -73,7 +73,7 @@ def unit(session):
 
 @nox.session(python=ALL_PYTHON[-1])
 def prerelease_deps(session):
-    """Run the unit test suite."""
+    """Run the unit test suite against pre-release versions of dependencies."""
 
     # Install test environment dependencies
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
@@ -81,8 +81,8 @@ def prerelease_deps(session):
     # Install the package without dependencies
     session.install('-e', '.', '--no-deps')
 
-    # Because we test the minimum dependency versions using the minimum Python
-    # version, the lowest python runtime that we test has a corresponding constraints
+    # We test the minimum dependency versions using the minimum Python
+    # version so the lowest python runtime that we test has a corresponding constraints
     # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
     # dependencies and extras.
     with open(

--- a/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py
+++ b/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py
@@ -206,7 +206,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -222,7 +222,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/credentials/noxfile.py
+++ b/tests/integration/goldens/credentials/noxfile.py
@@ -73,7 +73,7 @@ def unit(session):
 
 @nox.session(python=ALL_PYTHON[-1])
 def prerelease_deps(session):
-    """Run the unit test suite."""
+    """Run the unit test suite against pre-release versions of dependencies."""
 
     # Install test environment dependencies
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
@@ -81,8 +81,8 @@ def prerelease_deps(session):
     # Install the package without dependencies
     session.install('-e', '.', '--no-deps')
 
-    # Because we test the minimum dependency versions using the minimum Python
-    # version, the lowest python runtime that we test has a corresponding constraints
+    # We test the minimum dependency versions using the minimum Python
+    # version so the lowest python runtime that we test has a corresponding constraints
     # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
     # dependencies and extras.
     with open(

--- a/tests/integration/goldens/credentials/noxfile.py
+++ b/tests/integration/goldens/credentials/noxfile.py
@@ -15,6 +15,7 @@
 #
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
@@ -49,6 +50,7 @@ nox.sessions = [
     "docs",
     "blacken",
     "lint",
+    "prerelease_deps",
 ]
 
 @nox.session(python=ALL_PYTHON)
@@ -57,6 +59,80 @@ def unit(session):
 
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
     session.install('-e', '.')
+
+    session.run(
+        'py.test',
+        '--quiet',
+        '--cov=google/iam/credentials_v1/',
+        '--cov=tests/',
+        '--cov-config=.coveragerc',
+        '--cov-report=term',
+        '--cov-report=html',
+        os.path.join('tests', 'unit', ''.join(session.posargs))
+    )
+
+@nox.session(python=ALL_PYTHON[-1])
+def prerelease_deps(session):
+    """Run the unit test suite."""
+
+    # Install test environment dependencies
+    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
+
+    # Install the package without dependencies
+    session.install('-e', '.', '--no-deps')
+
+    # Because we test the minimum dependency versions using the minimum Python
+    # version, the lowest python runtime that we test has a corresponding constraints
+    # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
+    # dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{ALL_PYTHON[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    constraints_deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    session.install(*constraints_deps)
+
+    prerel_deps = [
+        "googleapis-common-protos",
+        "google-api-core",
+        "google-auth",
+        "grpcio",
+        "grpcio-status",
+        "protobuf",
+        "proto-plus",
+    ]
+
+    for dep in prerel_deps:
+        session.install("--pre", "--no-deps", "--upgrade", dep)
+
+    # Remaining dependencies
+    other_deps = [
+        "requests",
+    ]
+    session.install(*other_deps)
+
+    # Print out prerelease package versions
+
+    session.run("python", "-c", "import google.api_core; print(google.api_core.__version__)")
+    session.run("python", "-c", "import google.auth; print(google.auth.__version__)")
+    session.run("python", "-c", "import grpc; print(grpc.__version__)")
+    session.run(
+        "python", "-c", "import google.protobuf; print(google.protobuf.__version__)"
+    )
+    session.run(
+        "python", "-c", "import proto; print(proto.__version__)"
+    )
 
     session.run(
         'py.test',

--- a/tests/integration/goldens/credentials/tests/unit/gapic/credentials_v1/test_iam_credentials.py
+++ b/tests/integration/goldens/credentials/tests/unit/gapic/credentials_v1/test_iam_credentials.py
@@ -196,7 +196,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -212,7 +212,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/eventarc/noxfile.py
+++ b/tests/integration/goldens/eventarc/noxfile.py
@@ -73,7 +73,7 @@ def unit(session):
 
 @nox.session(python=ALL_PYTHON[-1])
 def prerelease_deps(session):
-    """Run the unit test suite."""
+    """Run the unit test suite against pre-release versions of dependencies."""
 
     # Install test environment dependencies
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
@@ -81,8 +81,8 @@ def prerelease_deps(session):
     # Install the package without dependencies
     session.install('-e', '.', '--no-deps')
 
-    # Because we test the minimum dependency versions using the minimum Python
-    # version, the lowest python runtime that we test has a corresponding constraints
+    # We test the minimum dependency versions using the minimum Python
+    # version so the lowest python runtime that we test has a corresponding constraints
     # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
     # dependencies and extras.
     with open(

--- a/tests/integration/goldens/eventarc/noxfile.py
+++ b/tests/integration/goldens/eventarc/noxfile.py
@@ -15,6 +15,7 @@
 #
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
@@ -49,6 +50,7 @@ nox.sessions = [
     "docs",
     "blacken",
     "lint",
+    "prerelease_deps",
 ]
 
 @nox.session(python=ALL_PYTHON)
@@ -57,6 +59,80 @@ def unit(session):
 
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
     session.install('-e', '.')
+
+    session.run(
+        'py.test',
+        '--quiet',
+        '--cov=google/cloud/eventarc_v1/',
+        '--cov=tests/',
+        '--cov-config=.coveragerc',
+        '--cov-report=term',
+        '--cov-report=html',
+        os.path.join('tests', 'unit', ''.join(session.posargs))
+    )
+
+@nox.session(python=ALL_PYTHON[-1])
+def prerelease_deps(session):
+    """Run the unit test suite."""
+
+    # Install test environment dependencies
+    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
+
+    # Install the package without dependencies
+    session.install('-e', '.', '--no-deps')
+
+    # Because we test the minimum dependency versions using the minimum Python
+    # version, the lowest python runtime that we test has a corresponding constraints
+    # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
+    # dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{ALL_PYTHON[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    constraints_deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    session.install(*constraints_deps)
+
+    prerel_deps = [
+        "googleapis-common-protos",
+        "google-api-core",
+        "google-auth",
+        "grpcio",
+        "grpcio-status",
+        "protobuf",
+        "proto-plus",
+    ]
+
+    for dep in prerel_deps:
+        session.install("--pre", "--no-deps", "--upgrade", dep)
+
+    # Remaining dependencies
+    other_deps = [
+        "requests",
+    ]
+    session.install(*other_deps)
+
+    # Print out prerelease package versions
+
+    session.run("python", "-c", "import google.api_core; print(google.api_core.__version__)")
+    session.run("python", "-c", "import google.auth; print(google.auth.__version__)")
+    session.run("python", "-c", "import grpc; print(grpc.__version__)")
+    session.run(
+        "python", "-c", "import google.protobuf; print(google.protobuf.__version__)"
+    )
+    session.run(
+        "python", "-c", "import proto; print(proto.__version__)"
+    )
 
     session.run(
         'py.test',

--- a/tests/integration/goldens/eventarc/tests/unit/gapic/eventarc_v1/test_eventarc.py
+++ b/tests/integration/goldens/eventarc/tests/unit/gapic/eventarc_v1/test_eventarc.py
@@ -216,7 +216,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -232,7 +232,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/logging/noxfile.py
+++ b/tests/integration/goldens/logging/noxfile.py
@@ -73,7 +73,7 @@ def unit(session):
 
 @nox.session(python=ALL_PYTHON[-1])
 def prerelease_deps(session):
-    """Run the unit test suite."""
+    """Run the unit test suite against pre-release versions of dependencies."""
 
     # Install test environment dependencies
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
@@ -81,8 +81,8 @@ def prerelease_deps(session):
     # Install the package without dependencies
     session.install('-e', '.', '--no-deps')
 
-    # Because we test the minimum dependency versions using the minimum Python
-    # version, the lowest python runtime that we test has a corresponding constraints
+    # We test the minimum dependency versions using the minimum Python
+    # version so the lowest python runtime that we test has a corresponding constraints
     # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
     # dependencies and extras.
     with open(

--- a/tests/integration/goldens/logging/noxfile.py
+++ b/tests/integration/goldens/logging/noxfile.py
@@ -15,6 +15,7 @@
 #
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
@@ -49,6 +50,7 @@ nox.sessions = [
     "docs",
     "blacken",
     "lint",
+    "prerelease_deps",
 ]
 
 @nox.session(python=ALL_PYTHON)
@@ -57,6 +59,80 @@ def unit(session):
 
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
     session.install('-e', '.')
+
+    session.run(
+        'py.test',
+        '--quiet',
+        '--cov=google/cloud/logging_v2/',
+        '--cov=tests/',
+        '--cov-config=.coveragerc',
+        '--cov-report=term',
+        '--cov-report=html',
+        os.path.join('tests', 'unit', ''.join(session.posargs))
+    )
+
+@nox.session(python=ALL_PYTHON[-1])
+def prerelease_deps(session):
+    """Run the unit test suite."""
+
+    # Install test environment dependencies
+    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
+
+    # Install the package without dependencies
+    session.install('-e', '.', '--no-deps')
+
+    # Because we test the minimum dependency versions using the minimum Python
+    # version, the lowest python runtime that we test has a corresponding constraints
+    # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
+    # dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{ALL_PYTHON[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    constraints_deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    session.install(*constraints_deps)
+
+    prerel_deps = [
+        "googleapis-common-protos",
+        "google-api-core",
+        "google-auth",
+        "grpcio",
+        "grpcio-status",
+        "protobuf",
+        "proto-plus",
+    ]
+
+    for dep in prerel_deps:
+        session.install("--pre", "--no-deps", "--upgrade", dep)
+
+    # Remaining dependencies
+    other_deps = [
+        "requests",
+    ]
+    session.install(*other_deps)
+
+    # Print out prerelease package versions
+
+    session.run("python", "-c", "import google.api_core; print(google.api_core.__version__)")
+    session.run("python", "-c", "import google.auth; print(google.auth.__version__)")
+    session.run("python", "-c", "import grpc; print(grpc.__version__)")
+    session.run(
+        "python", "-c", "import google.protobuf; print(google.protobuf.__version__)"
+    )
+    session.run(
+        "python", "-c", "import proto; print(proto.__version__)"
+    )
 
     session.run(
         'py.test',

--- a/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_config_service_v2.py
+++ b/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_config_service_v2.py
@@ -195,7 +195,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -211,7 +211,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_logging_service_v2.py
+++ b/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_logging_service_v2.py
@@ -196,7 +196,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -212,7 +212,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_metrics_service_v2.py
+++ b/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_metrics_service_v2.py
@@ -194,7 +194,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -210,7 +210,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/integration/goldens/redis/noxfile.py
+++ b/tests/integration/goldens/redis/noxfile.py
@@ -73,7 +73,7 @@ def unit(session):
 
 @nox.session(python=ALL_PYTHON[-1])
 def prerelease_deps(session):
-    """Run the unit test suite."""
+    """Run the unit test suite against pre-release versions of dependencies."""
 
     # Install test environment dependencies
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
@@ -81,8 +81,8 @@ def prerelease_deps(session):
     # Install the package without dependencies
     session.install('-e', '.', '--no-deps')
 
-    # Because we test the minimum dependency versions using the minimum Python
-    # version, the lowest python runtime that we test has a corresponding constraints
+    # We test the minimum dependency versions using the minimum Python
+    # version so the lowest python runtime that we test has a corresponding constraints
     # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
     # dependencies and extras.
     with open(

--- a/tests/integration/goldens/redis/noxfile.py
+++ b/tests/integration/goldens/redis/noxfile.py
@@ -15,6 +15,7 @@
 #
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
@@ -49,6 +50,7 @@ nox.sessions = [
     "docs",
     "blacken",
     "lint",
+    "prerelease_deps",
 ]
 
 @nox.session(python=ALL_PYTHON)
@@ -57,6 +59,80 @@ def unit(session):
 
     session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
     session.install('-e', '.')
+
+    session.run(
+        'py.test',
+        '--quiet',
+        '--cov=google/cloud/redis_v1/',
+        '--cov=tests/',
+        '--cov-config=.coveragerc',
+        '--cov-report=term',
+        '--cov-report=html',
+        os.path.join('tests', 'unit', ''.join(session.posargs))
+    )
+
+@nox.session(python=ALL_PYTHON[-1])
+def prerelease_deps(session):
+    """Run the unit test suite."""
+
+    # Install test environment dependencies
+    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
+
+    # Install the package without dependencies
+    session.install('-e', '.', '--no-deps')
+
+    # Because we test the minimum dependency versions using the minimum Python
+    # version, the lowest python runtime that we test has a corresponding constraints
+    # file, located at `testing/constraints-<version>-.txt`,  which contains all of the
+    # dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{ALL_PYTHON[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    constraints_deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    session.install(*constraints_deps)
+
+    prerel_deps = [
+        "googleapis-common-protos",
+        "google-api-core",
+        "google-auth",
+        "grpcio",
+        "grpcio-status",
+        "protobuf",
+        "proto-plus",
+    ]
+
+    for dep in prerel_deps:
+        session.install("--pre", "--no-deps", "--upgrade", dep)
+
+    # Remaining dependencies
+    other_deps = [
+        "requests",
+    ]
+    session.install(*other_deps)
+
+    # Print out prerelease package versions
+
+    session.run("python", "-c", "import google.api_core; print(google.api_core.__version__)")
+    session.run("python", "-c", "import google.auth; print(google.auth.__version__)")
+    session.run("python", "-c", "import grpc; print(grpc.__version__)")
+    session.run(
+        "python", "-c", "import google.protobuf; print(google.protobuf.__version__)"
+    )
+    session.run(
+        "python", "-c", "import proto; print(proto.__version__)"
+    )
 
     session.run(
         'py.test',

--- a/tests/integration/goldens/redis/tests/unit/gapic/redis_v1/test_cloud_redis.py
+++ b/tests/integration/goldens/redis/tests/unit/gapic/redis_v1/test_cloud_redis.py
@@ -207,7 +207,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [int(part) for part in google.auth.__version__.split(".")]
+    google_auth_major, google_auth_minor = [int(part) for part in google.auth.__version__.split(".")[0:2]]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials = ga_credentials.AnonymousCredentials()
         credentials._universe_domain = "foo.com"
@@ -223,7 +223,7 @@ def test__validate_universe_domain(client_class, transport_class, transport_name
         #
         # TODO: Make this test unconditional once the minimum supported version of
         # google-api-core becomes 2.15.0 or higher.
-        api_core_major, api_core_minor, _ = [int(part) for part in api_core_version.__version__.split(".")]
+        api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
         if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 15):
             client = client_class(client_options={"universe_domain": "bar.com"}, transport=transport_class(credentials=ga_credentials.AnonymousCredentials(),))
             with pytest.raises(ValueError) as excinfo:

--- a/tests/system/conftest.py
+++ b/tests/system/conftest.py
@@ -144,8 +144,8 @@ def parametrized_echo(use_mtls, channel_creator, transport_name, transport_endpo
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [
-        int(part) for part in google.auth.__version__.split(".")
+    google_auth_major, google_auth_minor = [
+        int(part) for part in google.auth.__version__.split(".")[0:2]
     ]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         credentials._universe_domain = credential_universe

--- a/tests/system/test_universe_domain.py
+++ b/tests/system/test_universe_domain.py
@@ -41,8 +41,8 @@ def test_universe_domain_validation_pass(parametrized_echo, channel_creator, tra
     # TODO: This is needed to cater for older versions of google-auth
     # Make this test unconditional once the minimum supported version of
     # google-auth becomes 2.23.0 or higher.
-    google_auth_major, google_auth_minor, _ = [
-        int(part) for part in google.auth.__version__.split(".")
+    google_auth_major, google_auth_minor = [
+        int(part) for part in google.auth.__version__.split(".")[0:2]
     ]
     if google_auth_major > 2 or (google_auth_major == 2 and google_auth_minor >= 23):
         assert parametrized_echo.transport._credentials.universe_domain == credential_universe

